backend/controller/dal/lease.go

```diff
@@ -23,7 +25,8 @@ type Lease struct {
 	ttl            time.Duration
 	errch          chan error
 	release        chan bool
-	leak           bool // For testing.
+	cancelCtx      context.CancelFunc // Cancels context created for lease owner.
```

Comment:
This doesn't need to be a field, just pass the context through to `renew()` directly